### PR TITLE
Update to 1.21

### PR DESCRIPTION
**Pull request by theEvilReaper**
## Overview

The pull request updates the data generator to the Minecraft version 1.21

### CODE REVIEW

File: DataGenerator/src/main/java/net/minestom/generators/DyeColorGenerator.java [hunk 1]
```diff
@@ -14,7 +14,7 @@ public JsonArray generate() {
             JsonObject dyeColor = new JsonObject();
             dyeColor.addProperty("id", dC.getId());
             dyeColor.addProperty("name", dC.name());
-            dyeColor.addProperty("textureDiffuseColor", convertTextureDiffuseColors(dC.getTextureDiffuseColors()));
+            dyeColor.addProperty("textureDiffuseColor", dC.getTextureDiffuseColor() & 0xFFFFFF);
```
Comment (TheMeinerLP) [inline DataGenerator/src/main/java/net/minestom/generators/DyeColorGenerator.java:17]: Is that right ?

File: DataGenerator/src/main/java/net/minestom/generators/EntityGenerator.java
Comment (TheMeinerLP) [inline DataGenerator/src/main/java/net/minestom/generators/EntityGenerator.java:1]: This magical values look super weird. This class needs to be replaced

File: DataGenerator/src/main/java/net/minestom/generators/FeatureFlagGenerator.java
Comment (TheMeinerLP) [inline DataGenerator/src/main/java/net/minestom/generators/FeatureFlagGenerator.java:1]: This generation is now different as my one before

**Review by TheMeinerLP — CHANGES_REQUESTED** [review]

File: DataGenerator/src/main/java/net/minestom/generators/FeatureFlagGenerator.java
Comment (theEvilReaper) [reply DataGenerator/src/main/java/net/minestom/generators/FeatureFlagGenerator.java:1]: So what is the difference?

Comment (TheMeinerLP) [reply DataGenerator/src/main/java/net/minestom/generators/FeatureFlagGenerator.java:1]: Its generating IDs but Flags don't need them

File: DataGenerator/src/main/java/net/minestom/generators/EntityGenerator.java
Comment (theEvilReaper) [reply DataGenerator/src/main/java/net/minestom/generators/EntityGenerator.java:1]: The values are not so mysterious when you take a closer look at the [Motion of Entities](https://minecraft.wiki/w/Entity#Motion_of_entities) section from the game's wiki. I'm not sure if Mojang's code includes these values, but if it does, we can retrieve them from their code

**Review by TheMeinerLP — APPROVED** [review]

**Pull request merged**